pandas_gbq/gbq.py

```diff
@@ -7,9 +7,17 @@
 import time
 import warnings
 from datetime import datetime
+import typing
+from typing import Dict, Optional, Union
 
 import numpy as np
 
+# Only import at module-level at type checking time to avoid circular
+# dependencies in the pandas package, which has an optional dependency on
+# pandas-gbq.
+if typing.TYPE_CHECKING:
+    import pandas
```

Comment:
This will never be executed during the tests (`TYPE_CHECKING` is `False` at runtime), thus at some point coverage will complain. Let's add a `# pragma: NO COVER` comment to ignore this block.